array_api_compat/numpy/_aliases.py

```diff
@@ -77,20 +74,21 @@ def _supports_buffer_protocol(obj):
 # rather than trying to combine everything into one function in common/
 def asarray(
     obj: Union[
-        ndarray,
+        Array,
         bool,
         int,
         float,
-        NestedSequence[bool | int | float],
+        complex,
+        NestedSequence[bool | int | float | complex],
         SupportsBufferProtocol,
     ],
```

Comment:
```
    obj: complex | NestedSequence[complex] | SupportsBufferProtocol,
```

assuming that arrays are required to implement the buffer protocol

Reply:
They are not, because it's CPU-only.
It is explained here: https://data-apis.org/array-api/latest/design_topics/data_interchange.html

Reply:
Ok, good to know 👌🏻